articles/zk-snarks-under-the-hood/zk_snarks_under_the_hood.md

```diff
@@ -0,0 +1,139 @@
+---
+Title: "Zk-SNARKs: Under the Hood"
+Status: Done
+Level: 6
+Note: There is a few places that are hard to fully understand
+---
+
+# Introduction
+
+Remember that the solution to a QAP is a set of polynomials (A, B, C) such that A(x) * B(x) - C(x) = H(x) * Z(x), where:
+
+- A is a linear combination of a set of polynomials {A_1…A_m}
+- B is the linear combination of {B_1…B_m} with the same coefficients
+- C is a linear combination of {C_1…C_m} with the same coefficients 
+
+Coefficients are $1, x, x^2, x^3, ... , x^m$
+
+Note that from this step, we are going to use [elliptic curve](../../terms/elliptic_curve.md), so we will have a generator point G (corresponding to 1 in the elliptic curve) appear in our equation as a way to convert a normal number into an elliptic curve.
+
+However, in a large system with thousands of logic gates like a hash function or something similar, the list that the prover needs to send to the verifier would be too long. So, we use [knowledge of exponent](../../terms/knowledge_of_exponent.md) assumption to make it shorter.
+
+On the other hand, using the prefixed function Z = $(x-1)(x-2)(x-3)(x-4)$ as a divider for the prover's QAP is not secure enough as it would create room for the prover to make "fake" proofs.
```

Comment:
On the other hand ?

Reply:
It's the same as "However"

Reply:
![Uploading Screenshot 2024-01-06 at 18.47.05.png…]()

Reply:
<img width="623" alt="Screenshot 2024-01-06 at 18 50 09" src="https://github.com/sota-zk-lab/zkp-documents/assets/33181397/9d5803bf-8482-49ad-9b03-2a6292434238">

Reply:
Uh, this should be the presenting alternative meaning of that phrase 😅